tests/ConfigR.Tests.Acceptance.Roslyn.CSharp/DynamicReferencesFeature.cs

```diff
@@ -4,122 +4,129 @@
         {
-            "Given a local config file adds a Foo from a dynamically loaded assembly with a Bar of 'baz'"
-                .f(() =>
+            dynamic config = null;
+
+            "Given an assembly in memory which defines a Foo type"
+                .f(c =>
                 {
-                    using (var writer = new StreamWriter(LocalScriptFileConfig.Path))
+                    var code =
+$@"namespace {c.Step.Scenario.ScenarioOutline.Method.Name}
+{{
+    public class Foo
+    {{
+        public string Bar {{ get; set; }}
+    }}
+}}";
+
+                    var compilation = CSharpCompilation.Create(
+                            c.Step.Scenario.ScenarioOutline.Method.Name,
+                            new[] { CSharpSyntaxTree.ParseText(code) },
+                            null,
+                            new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
+                        .AddReferences(MetadataReference.CreateFromFile(typeof(object).Assembly.Location));
+
+                    metadataReference = compilation.ToMetadataReference();
+
+                    using (var stream = new MemoryStream())
                     {
-                        writer.WriteLine(@"using ConfigR.Tests.Support.SampleDependency;");
-                        writer.WriteLine(@"Add(""foo"", new Foo { Bar = ""baz"" });");
-                        writer.Flush();
+                        var emitResult = compilation.Emit(stream);
+                        emitResult.Success.Should().BeTrue(
+                            "The diagnostics should indicate no errors:{0}{1}",
+                            Environment.NewLine,
+                            string.Join(Environment.NewLine, emitResult.Diagnostics.Select(diagnostic => diagnostic.ToString())));
+
+                        stream.Seek(0, SeekOrigin.Begin);
+                        reference = Assembly.Load(stream.ToArray());
                     }
-                })
-                .Teardown(() => File.Delete(LocalScriptFileConfig.Path));
+                });
 
-            "When I load the assembly"
-                .f(() => reference = Assembly.Load("ConfigR.Tests.Support.SampleDependency"));
+            "And config file with Foo with a Bar of 'baz'"
+                .f(c =>
+                {
+                    var code =
+$@"using {c.Step.Scenario.ScenarioOutline.Method.Name};
+Config.Foo = new Foo {{ Bar = ""baz"" }};
+";
+                    ConfigFile.Create(code).Using(c);
+                });
 
-            "And I add the reference to global autoloading"
-                .f(() => Config.GlobalAutoLoadingReferences.Add(reference));
+            "And I load the config using the assembly as a reference"
+                .f(async () => config = await new Config().UseRoslynCSharpLoader()
+                    .AddReferences(reference)
+                    .AddReferences(metadataReference)
+                    .Load());

```

Comment:
never mind, I'm going with a different approach which exposes `ScriptOptions` and `InteractiveAssemblyLoader` to the consumer, so what they actually do with it isn't ConfigR's concern.